src/System.Management.Automation/engine/remoting/server/ServerRunspacePoolDriver.cs

```diff
@@ -1392,126 +1423,102 @@ private class DebuggerCommandArgument
                 // Input parameters:
                 // [-Id <int>]
                 // Returns Breakpoint object(s).
-                string script = null;
-
-                if (command.Parameters?.Count > 0)
+                TryGetParameter<int?>(command, "RunspaceId", out int? runspaceId);
+                if (TryGetParameter<int>(command, "Id", out int breakpointId))
                 {
-                    int breakpointId = CheckBreakpointIdParameter(command);
-                    script = $"$host.Runspace.Debugger.GetBreakpoint({breakpointId})";
+                     preProcessOutput.Add(serverRemoteDebugger.GetBreakpoint(breakpointId, runspaceId));
                 }
                 else
                 {
-                    script = $"$host.Runspace.Debugger.GetBreakpoints()";
+                    foreach (Breakpoint breakpoint in serverRemoteDebugger.GetBreakpoints(runspaceId))
+                    {
+                        preProcessOutput.Add(breakpoint);
+                    }
                 }
 
-                ReplaceVirtualCommandWithScript(commands, script);
+                result = PreProcessCommandResult.BreakpointManagement;
             }
             else if (commandText.Equals(RemoteDebuggingCommands.SetBreakpoint, StringComparison.OrdinalIgnoreCase))
             {
                 // __Set-PSBreakpoint private virtual command.
                 // Input parameters:
-                // [-Script] <string[]> [-Line] <int[]> [[-Column] <int>] [-Action <scriptblock>]
-                // [[-Script] <string[]>] -Command <string[]> [-Action <scriptblock>]
-                // [[-Script] <string[]>] -Variable <string[]> [-Action <scriptblock>] [-Mode <VariableAccessMode>]
-                // Returns Breakpoint object.
-                if (command.Parameters == null || command.Parameters.Count == 0)
+                // -Breakpoint <Breakpoint> or -BreakpointList <IEnumerable<Breakpoint>>
+                // [-RunspaceId <int?>]
+                // Returns Breakpoint object(s).
+                TryGetParameter<Breakpoint>(command, "Breakpoint", out Breakpoint breakpoint);
+                TryGetParameter<IEnumerable<Breakpoint>>(command, "BreakpointList", out IEnumerable<Breakpoint> breakpoints);
+                if (breakpoint == null && breakpoints == null)
```

Comment:
maybe also check breakpoints.Count > 0

Reply:
I don't think we want to throw if someone passes in an empty list to `SetBreakpoints()`